sample/src/main/java/com/robotemi/sdk/sample/MainActivity.java

```diff
@@ -0,0 +1,313 @@
+    }
+
+    /*
+        This is an example of saving locations.
+     */
+
+    public void saveLocation(View view){
+        String location = etSaveLocation.getText().toString().toLowerCase().trim();
+        robot.saveLocation(location);
+        robot.speak(TtsRequest.create("I've successfully saved the " + location + " location.", true));
+        hideKeyboard(MainActivity.this);
+    }
+
+    /*
+        goTo checks that the location sent is saved then goes to that location.
+     */
+
+    public void goTo(View view) {
+        for (String location : robot.getLocations()){
+            if (location.equals(etGoTo.getText().toString().toLowerCase().trim())) {
+                robot.goTo(etGoTo.getText().toString().toLowerCase().trim());
+                hideKeyboard(MainActivity.this);
+            }
+        }
+    }
+
+    /*
+        Simple follow me example.
+     */
+
+    public void followMe(View view) {
+        robot.beWithMe();
+        hideKeyboard(MainActivity.this);
+    }
+
+    /*
+        Manually navigate the robot with skidJoy, tiltAngle, turnBy and tiltBy.
+     */
+
+    public void skidJoy(View view) {
+        robot.skidJoy(1.0F, 1.0F);
+    }
+
+    public void tiltAngle(View view) {
+        robot.tiltAngle(23, 5.3F);
+    }
+
+    public void turnBy(View view) {
+        robot.turnBy(90, 6.2F);
+    }
+
+    public void tiltBy(View view) {
+        robot.tiltBy(30, 1.2F);
+    }
+
+    /*
+        Hiding keyboard after every button press
+     */
+
+    public static void hideKeyboard(Activity activity) {
+        InputMethodManager imm = (InputMethodManager) activity.getSystemService(Activity.INPUT_METHOD_SERVICE);
+        //Find the currently focused view, so we can grab the correct window token from it.
+        View view = activity.getCurrentFocus();
+        //If no view currently has focus, create a new one, just so we can grab a window token from it
+        if (view == null) {
+            view = new View(activity);
+        }
+        imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
+    }
+
+    /*
+        Display the saved locations in a dialog
+     */
+
+    public void savedLocationsDialog(View view) {
+        hideKeyboard(MainActivity.this);
+        locations = robot.getLocations();
+        CustomAdapter customAdapter = new CustomAdapter(MainActivity.this, android.R.layout.simple_selectable_list_item, locations);
+        AlertDialog.Builder versionsDialog = new AlertDialog.Builder(MainActivity.this);
+        versionsDialog.setTitle("Saved Locations:");
+        versionsDialog.setPositiveButton("OK", null);
+        versionsDialog.setAdapter(customAdapter, null);
+        AlertDialog dialog = versionsDialog.create();
+        dialog.show();
+    }
+
+
+    // TODO: 2019-06-16 Work on onNlpCompleted. At the moment I couldn't get it to load.
+    @Override
+    public void onNlpCompleted(NlpResult nlpResult) {
+        //do something with nlp result. Base the action on what you get back
```

Comment:
Write something about those actions needs to be declared in the skill manifest

Reply:
I will write about what needs to be declared in the manifest, but what does the onNlpCompleted method have to do with it?